pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -909,6 +912,51 @@ private void checkHybridTableConfig(String rawTableName, TableConfig tableConfig
     }
   }
 
+  /*
+   * Validates that we have sufficient number of servers in the specified tenants to be able to serve the requested
+   * replication in the table config
+   */
+  private void ensureCanHostAllReplicas(TableConfig tableConfig) {
+    int replication = tableConfig.getReplication();
+    TagOverrideConfig tagOverrideConfig = tableConfig.getTenantConfig().getTagOverrideConfig();
+
+    if (tagOverrideConfig != null && tagOverrideConfig.getRealtimeConsuming() != null
+        && tagOverrideConfig.getRealtimeCompleted() != null) {
```

Comment:
Could you add a comment explaining the significance of these 3 references not being null?

Reply:
You might want to use `TagNameUtils`? I think we allow having only one of `CONSUMING/COMPLETED` not null